notebooks/Notebook1.md

```diff
@@ -18,17 +24,35 @@ exciting milestones, preliminary results, etc. Feel free to include images
 code, and any other resources that you think will help clearly convey your
 design process.
 
+This week has been a very interesting, design-wise. For the flexibility criterion I discussed above, I decided very quickly that my project should be an internal DSL. That way, in case there is some way in which the language is incomplete, a seasoned programmer could add some host language code as a stopgap until the DSL can be updated. This realization was difficult- I spent a few hours programming an object-oriented framework for board games, then extended it to 52-card-deck card games, then used that framework to program the game Hearts. The process of programming exposed a lot of the "sticking points" where a DSL could be useful. For example, many games have a set round structure, in which each player takes a turn; each of these turns is a specific sequence of steps, each of which are made of one or more actions. These actions are in turn composed of atomic moves, like moving a card from one location to another, or rolling dice. The below document outlines the current OOP structure of game objects, but but the Round-Turn-Action-Move relationship is currently eluding me.
+
+I implemented everything this week in Python, as it was quick and simple to get my ideas down, which let me find the places where a DSL could really fill in the gaps. However, I'm still confident that I want to implement the actual DSL in Scala, as I think the features it offers, both in terms of customizability and OOP in general, will allow for a better-constructed language. This is a bit of a tradeoff, however. Having a DSL hosted in Scala rather than in Python means that a prospective programmer will have a larger barrier to entry. As I alluded to in my first section, though, this will likely mean that the language is easier to read, but harder to write (at least, for someone with little CS experience). Since I consider the functionality and readability of the language to be more important than its writeability, I think this is a tradeoff worth making. While I'm a bit disappointed that the language will not be as easy to use as I had originally intended, I think this is in the language's best interest. With many more weeks of development, I feel confident that a GUI could be devised, perhaps Scratch-style, that would make development in this DSL as easy as I want it to be, without sacrificing its flexibility or readability.
```

Comment:
I think a gui, would not be ideal for your project. Focus on creating a solid *internal* DSL and change your audience away from the "newbie." Just make your expectation a little bit of coding experience. Allow it to be a preset chunk of tools and resources for anyone making a board game simulation.

Reply:
Basically focus on usability but not intuitively. Of course, this is just my opinion. But I think a GUI may not be ideal for your project.